server/src/main/java/org/apache/druid/server/QueryResultPusher.java

```diff
@@ -278,6 +274,17 @@ private Response handleDruidException(ResultsWriter resultsWriter, DruidExceptio
     }
   }
 
+  private DruidException toDruidException(Exception e)
+  {
+    if (e instanceof DruidException) {
+      return (DruidException) e;
+    }
+    if (!(e instanceof QueryException)) {
+      e = new QueryInterruptedException(e);
+    }
+    return DruidException.fromFailure(new QueryExceptionCompat((QueryException) e));
```

Comment:
(QueryException) is it safe cast?

Reply:
QueryInterruptedException extends QueryException. So should be good

Reply:
yes - actually; I've changed the way this is fixed

I now think that it was not intentional that `QueryExceptionCompat` have not passed the `exception`  ; only it's message

cc: @cheddar